routes/saveRoute.js

```diff
@@ -1,34 +1,53 @@
 import { Router } from "express";
-import { Routes } from "../firebase.js"
+import admin from "firebase-admin";
+import { Routes } from "../firebase.js";
+import { User } from "../firebase.js";
 
+// Save new routes
 const router = Router();
 
-const save = async (userName, routesGeometry, distance, time, likes) => {
+const addToRoute = async (username, timestamp, routeGeometry, distance, duration, likes) => {
   await Routes.add({
-    Username: userName,
-    Geometry: routesGeometry,
+    Username: username,
+    Timestamp: timestamp,
+    Geometry: routeGeometry,
     Distance: distance,
-    Time: time,
+    Duration: duration,
     Likes: likes
   });
 }
 
+const addToUser = async (username, routeGeometry) => {
+  const user = await User.doc(username).get();
+  if (user.exists) {
+    const userData = user.data();
+    const routes = userData.Routes;
+    routes.push(routeGeometry);
+    await User.doc(username).update({ Routes: routes })
+  }
+  else {
+    console.log("Unable to add to User")
+  }
+}
+
```

Comment:
Should we throw an error here instead of console logging ah, so
```
if (user.exists) {
  ...
} else {
  throw new Error("Unable to add to User")
}
```